test/properties.spec.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+var gentest = require('gentest');
+var t = gentest.types;
+var chai = require('chai');
+var expect = chai.expect;
+
+var encode = require('../src/encoder');
+var decode = require('../src/decoder');
+var tt = require('../src/type-tags');
+
+
+var SAMPLE_SIZE = 10;
+
+describe('properties', function () {
+
+  describe('integers should always encode to the same value', function () {
+    var generator = t.int;
+    gentest.sample(generator, SAMPLE_SIZE).forEach(function (a) {
+      it('should deterministically encode ' + JSON.stringify(a), function () {
+        expect(encode(a)).to.eql(encode(a));
+      });
+    });
+  });
+
+  describe('integers should round-trip', function () {
+    var generator = t.int;
+    gentest.sample(generator, SAMPLE_SIZE).forEach(function (a) {
+      it('should round-trip ' + JSON.stringify(a), function () {
+        expect(decode(encode(a))).to.eql(a);
+      });
+    });
+  });
+
+  describe('arrays should have the same length after round-tripping', function () {
+    var generator = t.arrayOf(t.int);
+    gentest.sample(generator, SAMPLE_SIZE).forEach(function (a) {
+      it('should preserve length during round-trip of ' + JSON.stringify(a), function () {
+        expect(decode(encode(a)).length).to.eql(a.length);
+      });
+    });
+  });
+
+  function isUint(typeTag) {
+    return (0b11000000 & typeTag) < 2 || typeTag >= 0xE4 && typeTag <= 0xE7;
```

Comment:
do we have these tags defined somewhere or do we have to use magic numbers in the test?